base/src/main/java/org/aya/concrete/stmt/Decl.java

```diff
@@ -113,8 +113,8 @@ public PrimDecl(
       this.ref = DefVar.concrete(this, name);
     }
 
-    @Override public boolean needTyck() {
-      return ref.concrete.signature == null;
+    @Override public boolean needTyck(@NotNull ImmutableSeq<String> currentMod) {
+      return ref.isInModule(currentMod) && ref.concrete.signature == null;
```

Comment:
Hmmm, so prims will be considered need-tyck in the current module. Is that desirable?

Reply:
> prims will be considered need-tyck in the current module 

because it may have concrete tele and result (see StmtTycker).

Reply:
And you may misunderstand the `currentModule`. It means the module who are using this `DefVar`

Reply:
👌🏼